### ISSUE TITLE
Error when using a component in a mailer view

### ISSUE THREAD

**Issue by dark-panda** [github/view_component#940]
The following commit broke the use of components in mailers:

ab77a208aff82c357de513c1407b275c512c3d40

Mailers don't have a `request`, and when we try to do `@request ||= controller.request` we get errors like

```
undefined method `request' for #<InvitationMailer:0x0000000003a638>
```

This only happens when you use something that needs to reference the `request` in the mailer, like `asset_url` or `image_tag` or whatever. 

An easy fix is to change the offending line to

```ruby
@request ||= controller.try(:request)

# or

@request ||= controller.request if controller.respond_to?(:request)
```

### Steps to reproduce

Use a component in an ActionMailer instance where `compute_asset_host` is necessary, like using `asset_url` or `image_tag`.

### Expected behavior

It should not attempt to use `request` if `controller` does not respond to it, where in this case `controller` is an instance of an `ActionMailer`.

### Actual behavior

It errors out with an `undefined method `request'`.

### System configuration

**Rails version**: 6.1.3.2

**Ruby version**: 2.7.1

**Gem version**: 2.2.16

**Comment by BlakeWilliams** [comment]
👋 Thanks for opening this!

We should definitely support view components use in mailers. Would you mind opening a PR with a failing test for us to investigate? Alternatively, if you would like to open a PR with a failing test and a fix, that would be awesome too. 🙂

**Comment by Spone** [comment]
Fixed by #948
